### ISSUE TITLE
[7.6][build] License checker is fragile, breaking release manager builds. 

### ISSUE THREAD

**Issue by robbavey** [elastic/logstash#11796]
The `7.6` branch of logstash still has the version check in place for license dependencies removed from later branches [here](https://github.com/elastic/logstash/pull/11073)

This has the problem of causing issues with causing [failures in release manager builds](https://internal-ci.elastic.co/job/elastic+release-manager+7.6+unified-snapshot/113) when dependencies change.

**Comment by jsvd** [comment]
Although >= 7.7 branches no longer care about dependency versions, there's an underlying problem that is present in all release branches since they have a release lockfile that points to the release version, for example ["logstash-core, 7.6.3"](https://github.com/elastic/logstash/blob/7.6/Gemfile.jruby-2.5.lock.release).
When running tasks like `./gradlew installDefaultGems`, if one doesn't call the command with `RELEASE=1`, the tasks will assume this is SNAPSHOT build, and bundler will invalidate the lockfile. We can see in debug mode when doing `DEBUG=1 ./gradlew clean installDefaultGems`:

```
Found changes from the lockfile, re-resolving dependencies because the gemspecs for path gems changed, the lockfile does not have all gems needed for the current platform
```

This results in the `Gemfile.lock` having different gems than `Gemfile.jruby-2.5.lock.release`:

```
❯ grep pry Gemfile*                           
Gemfile.jruby-2.5.lock.release:      pry (~> 0.12)
Gemfile.jruby-2.5.lock.release:    pry (0.13.0-java)
Gemfile.lock:      pry (~> 0.12)
Gemfile.lock:    pry (0.13.1-java)
```

This doesn't happen with `RELEASE=1 ./gradlew clean installDefaultGems`:

```
❯ grep pry Gemfile*                           
Gemfile.jruby-2.5.lock.release:      pry (~> 0.12)
Gemfile.jruby-2.5.lock.release:    pry (0.13.0-java)
Gemfile.lock:      pry (~> 0.12)
Gemfile.lock:    pry (0.13.1-java)
```

**Comment by jsvd** [comment]
closed by #11813
